personal_finances/transaction/filtering.py

```diff
@@ -16,6 +16,11 @@ def _filter_by_property_range(
 def transaction_datetime_filter(
     start: datetime, end: datetime, transactions: List[SimpleTransaction]
 ) -> List[SimpleTransaction]:
+    if start > end:
+        raise ValueError("Start date cannot be greater than end date")
+    if not transactions:
+        raise ValueError("Transaction list should not be empty")
```

Comment:
I made an exception in case of empty list for traceability purposes. If somehow the software send an empty list due to a bug, we can know as soon as possible. Seems to me that a no op here might could lead to an unexpected behavior, where we need to invest more time figuring out exactly what is happening. What your thoughts about it?

Reply:
As a general principle of single responsibility, I wouldn't worry about bugs coming from other parts of the code. Coupling this worry on a bigger scale can lead into a messy codebase.

But you're right that it's useful to trace things for debugability, I'd use logs for tracing in this case. You can log the input/output list size in this case.

Reply:
Another argument here is that the assumption that "empty list is a bug" is not correct. I can have period of time I selected that I don't have transactions and that's fine, it's not a good predictor of a bug.